### PR TITLE
eliminate multiple return statements for wasm2asm functions

### PR DESCRIPTION
**Pull request by froydnj**
This change eliminates one issue that prevents asm.js validation of the
generated code, see #1443.

### CODE REVIEW

File: src/wasm2asm.h [hunk 1]
```diff
@@ -655,7 +655,8 @@ Ref Wasm2AsmBuilder::processFunction(Function* func) {
   ExpressionList* stats = isBodyBlock ?
       &static_cast<Block*>(func->body)->list : nullptr;
   bool endsInReturn =
-      (isBodyBlock && ((*stats)[stats->size()-1]->_id == Expression::ReturnId));
+      (isBodyBlock && ((*stats)[stats->size()-1]->_id == Expression::ReturnId)) ||
```
Comment (kripken) [inline src/wasm2asm.h]: While you're here, please change `->_id == Expression::ReturnId` to `->is<Return>()`. (Also the same happens with `Block` a few lines above.)

**Comment by froydnj** [comment]
> Not sure what the CI errors are - maybe not all tests were updated?

Sigh, apparently CI checks out *extra* tests in a different repo and runs those, too?  How are those supposed to be updated in parallel?

**Comment by froydnj** [comment]
Specifically, `address.2asm.js` and `forward.2asm.js` were not updated, because those files don't exist in the binaryen repo, only in the testsuite repo.

**Comment by kripken** [comment]
Is that the `spec` repo? I do see `foward.wast` in there. I guess we already have some code to run the spec test suite through wasm2asm? (which I was thinking we'd want eventually, surprised it's already started)

Perhaps the only thing missing is to add a few lines to `./auto_update_tests.py` for updating those, as the inputs are in the `spec` test repo, but the outputs are in this one.

**Comment by froydnj** [comment]
Oh, I see, I don't have to update a separate repo, I just have to have the repo checked out, and `auto_update_tests.py` will take care of those automatically.  I was confused!

**Comment by kripken** [comment]
Last errors look like binaryen.js, which means we need to update the wasm2asm component in the JS port too. If you have emscripten set up `./build-js.sh` can do that. If not I can do it for you.

**Comment by froydnj** [comment]
@kripken I don't know where to go from here.  The test passes locally for me, and I think I've regenerated the JS files correctly.  Can you try regenerating the files?

**Comment by kripken** [comment]
I think the only problem left here might be the merge conflict. I merged in master and rebuilt [here](https://github.com/WebAssembly/binaryen/commits/temp) and it looks fine. We can just merge that branch in, or you can merge it into here, either way.

**Comment by froydnj** [comment]
Sigh, last thing to fix:

```
executing:  bin/asm2wasm /home/travis/build/WebAssembly/binaryen/test/two_sides.asm.js --trap-mode=clamp -O0
[PassRunner] running passes...
[PassRunner]   running pass: finalize-calls...        0.0004724 seconds.
[PassRunner]   (validating)
The job exceeded the maximum time limit for jobs, and has been terminated.
```

Is this a known potential problem?  I guess it's _possible_ that the change might have eliminated a return statement that was there before, and thus the test never returns, or something?

**Comment by kripken** [comment]
I've seen that before, it seemed like a random CI failure. Might be ok after a restart.

**Comment by froydnj** [comment]
> I've seen that before, it seemed like a random CI failure. Might be ok after a restart.

Hm, OK.  Had to fix another merge conflict anyway, so rebased and pushed again.  We'll see if CI is happy now.

**Comment by kripken** [comment]
Looks good! :)

**Pull request merged**